lib/ansible/plugins/strategy/free.py

```diff
@@ -118,7 +121,32 @@ def run(self, iterator, play_context):
                     display.debug("this host has work to do", host=host_name)
 
                     # check to see if this host is blocked (still executing a previous task)
-                    if host_name not in self._blocked_hosts or not self._blocked_hosts[host_name]:
+                    if (host_name not in self._blocked_hosts or not self._blocked_hosts[host_name]):
+
+                        display.debug("getting variables", host=host_name)
+                        task_vars = self._variable_manager.get_vars(play=iterator._play, host=host, task=task,
+                                                                    _hosts=self._hosts_cache,
+                                                                    _hosts_all=self._hosts_cache_all)
```

Comment:
Nice! Thanks for remembering this :)

Reply:
Yes, moved it up so I could use them for the templating of the value sooner, and I figured it was inside the if block anyway so it was most likely going to be tagged for work and there wasn't any major performance reduction in doing it sooner.